### ISSUE TITLE
Problems with big chunks

### ISSUE THREAD

**Issue by themasch** [themasch/node-ebml#34]
To encode the data size Matroska uses a flexible length encoding. These are basically unsigned ints with a length from 1 to 8 bytes or from 7 to 56 bits. See the spec for more information:  https://www.matroska.org/technical/specs/index.html#EBML_ex

These seem to have two issues in node-ebml currently: 

- [ ] Integers with more than eight bytes, or bigger than 2^56-2. These shouldn't exist in ebml, following the spec. 
- [ ] Integers bigger than 2^53-1 but smaller than 2^56: Supported by EBML but not by JavaScript

The first group should be our main concern, I have not seen that yet and we may be able to just throw an error if that happens. We should check for the "all 1"-special case thou. 

The second group is going to be a problem. We might need something like bigint or bignum for that, but we can not use bignum as a buffer size etc. I guess we need some real-world samples to figure out what could be done here. 

Issues related to this are #8 and #10.

**Comment by themasch** [comment]
Thinking about it again. 2^53 bits are about 1 Petabyte. I guess we can assume that there just are no chunks with a defined size of that size?

**Comment by bradisbell** [comment]
How bad are the performance issues when switching to some big number library?  If this library won't support more than 2^53, it seems reasonable, but should be well documented I think.

**Comment by themasch** [comment]
I don't think we really can use bignum here because we are using the read ints as buffer sizes and I highly doubt that you can have buffers bigger than Number.MAX_SAFE_INTEGER bytes. 

If it were just for putting numbers in the exported structs I would try it, and maybe we should do it for numbers that are not used as buffer sizes but afaik we cannot use BigNum (or something like that) as a buffer size.

**Comment by themasch** [comment]
Maybe we should use things like BigNum here: https://github.com/themasch/node-ebml/blob/master/lib/ebml/tools.js#L197

**Comment by jayands** [comment]
Okay, I don't know why I didn't think of this before, but I fired up Node (10.9.0). @themasch, you're correct in that
>  I highly doubt that you can have buffers bigger than `Number.MAX_SAFE_INTEGER` bytes

![node buffer max alloc](https://user-images.githubusercontent.com/1018901/44623926-5932e780-a892-11e8-9d0a-0f885fdb0a6e.png)

**Comment by jayands** [comment]
Hey, get this! There was a BigInt added to Node 10.7! See [this article][v8] for the post announcing it, as well as the TC39 proposal for it being at [Stage 3][tc39].

[v8]: https://v8project.blogspot.com/2018/05/bigint.html
[tc39]: https://github.com/tc39/proposal-bigint

This is huge, because it means that tools like Babel will *have* to provide polyfills for them natively if/when it reaches Stage 4.

**Comment by Kylart** [comment]
Hello,

Do we have any update on this? I'm building a torrent live streaming video player and I run into issue #8   so I'd really love any news 😄
